pkg/acceptance/bettertestspoc/config/model/view_model_ext.go

```diff
@@ -1,6 +1,81 @@
 package model
 
+import (
+	"encoding/json"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/collections"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-testing/config"
+)
+
+func (v *ViewModel) MarshalJSON() ([]byte, error) {
+	type Alias ViewModel
+	return json.Marshal(&struct {
+		*Alias
+		DependsOn []string `json:"depends_on,omitempty"`
+	}{
+		Alias:     (*Alias)(v),
+		DependsOn: v.DependsOn(),
```

Comment:
Why can't we just make this field exported in `ResourceModelMeta`? Same for other similar fields.

Reply:
We could technically but then metadata becomes something different than designed to be (it gets the second responsibility next to just keeping the meta info about the resource). What other similar fields you have in mind?

I would leave it for now - adding this to the readme point about the `depends_on` generation.